torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -8,12 +8,27 @@
 #include <torch/csrc/WindowsTorchApiMacro.h>
 
 #include <type_traits>
+#include <unordered_map>
 
 namespace torch {
 namespace jit {
 namespace fuser {
 namespace cuda {
 
+// Note, the uniqueness of the ide generated for a given input set is only local
+// to the instance of `InputsCodeLookup`.
+class InputsCodeLookup {
+ public:
+  // encode each unique input sets to an unique id;
+  size_t getCode(const at::ArrayRef<IValue>& inputs);
+
+ protected:
+  size_t current_id_ = 1;
+
+  // TODO: change this to a trie for efficiency;
+  std::unordered_map<std::string, size_t> inputs_code_lookup_;
```

Comment:
just curious, did this show up in the execution profile as a significant overhead? I'd be surprised in a trie ends up significantly faster than a hash map

Reply:
I will profile it before doing anything specific.

Just to clarify, I'm not worried about hash lookup time vs trie traversal. It's actually the string concatenation that is needed to construct the key that I'm concerned.